master/internal/trial.go

```diff
@@ -1096,17 +1096,29 @@ func (t *trial) terminated(ctx *actor.Context) {
 		return
 	}
 
-	if w, err := t.sequencer.Workload(); err != nil {
-		if !t.replaying {
-			if err := markWorkloadErrored(t.db, w); err != nil {
-				ctx.Log().Error(err)
+	var w workload.Workload
+	var err error
+	e := workload.Errored
+	switch {
+	case !t.sequencer.UpToDate():
+		w, err = t.sequencer.Workload()
+		if err != nil && !t.replaying {
+			if wErr := markWorkloadErrored(t.db, w); wErr != nil {
+				ctx.Log().
+					WithError(wErr).
+					Error("failed to mark workload errored when terminated")
 			}
 		}
+	case t.sequencer.PrecloseCheckpointWorkload() != nil:
+		w = *t.sequencer.PrecloseCheckpointWorkload()
+	default:
+		err = errors.New("trial terminated due to failure but had nothing to fail")
```

Comment:
Should this be a panic?  does it meet our qualifications of being a "purely internal logic error"?

Reply:
yeah I believe it does. it's a totally unrecoverable, unexpected state.